### PR TITLE
udn, egress, e2e: assert a pod can reach a service outside the cluster

### PR DESCRIPTION
**Pull request by maiqueb**
#### What this PR does and why is it needed

This PR adds an e2e test to assert a pod can reach an HTTP service running outside the kubernetes cluster.

To prove it, we run a containerized http server in the kind network, and access it from a pod started in the Kubernetes cluster.

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by flavio-fernandes** [comment]
hi @maiqueb ! Could you help me running this test? I created a kind cluster using  `kind.sh -ic -ds -mne`

but the e2e test gets skipped:

```
ill run 0 of 283 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
P [PENDING]
Multi Homing multiple pods connected to the same OVN-K secondary network localnet OVN-K secondary network with a service running on the underlay when a policy is provisioned can communicate over a localnet secondary network from pod to gw ingress denyall, ingress policy should have no impact on egress [BUG, OCPBUGS-25928]
/home/vagrant/dev/ovn-kubernetes.git/test/e2e/multihoming.go:842
------------------------------
SSSSSSSSSSSSSS
------------------------------
P [PENDING]
Multi Homing multiple pods connected to the same OVN-K secondary network multi-network policies multi-network ingress allow all for a localnet topology when the multi-net policy is egress deny-all, should not affect ingress [BUG, OCPBUGS-25928]
/home/vagrant/dev/ovn-kubernetes.git/test/e2e/multihoming.go:1395
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
P [PENDING]
Network Segmentation a user defined primary network an HTTP service which is deployed outside the Kubernetes cluster can be accessed to from the pods running in the Kubernetes cluster by one pod with a single IPv4 address
/home/vagrant/dev/ovn-kubernetes.git/test/e2e/network_segmentation.go:420
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
```

launch.json:

```
"version": "0.2.0",
    "configurations": [
        {
            "name": "e2e flaviof",
            "type": "go",
            "request": "launch",
            "mode": "test",
            "program": ".",
            "args": [
                "-ginkgo.focus",
                // "Network Segmentation",
                "an HTTP service which is deployed outside the Kubernetes cluster",
                "-ginkgo.flake-attempts",
                "1",
                "-ginkgo.fail-fast",
                "-provider",
                "skeleton",
                "-kubeconfig",
                "/home/vagrant/admin.conf",
            ],
            "env": {
                "KUBECONFIG": "/home/vagrant/admin.conf"
            }
        }
    ]
}
```

File: test/e2e/network_segmentation.go [hunk 1]
```diff
@@ -366,6 +366,75 @@ var _ = Describe("Network Segmentation", func() {
...
+						}
+						return updatedPod.Status.Phase
+					}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
+
+					By("asserting the *client* pod can contact the server located outside the cluster")
+					Eventually(func() error {
+						return connectToServer(clientPodConfig, externalIpv4, port)
+					}, 2*time.Minute, 6*time.Second).Should(Succeed())
+				},
+				Entry("by one pod with a single IPv4 address",
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply test/e2e/network_segmentation.go:427]: well, the plan is to add more entries later. 

This feature doesn't even exist yet. We're on it.

**Comment by maiqueb** [comment]
@flavio-fernandes the test is marked as pending in the code: https://github.com/ovn-org/ovn-kubernetes/pull/4515/files#diff-2be1e01ba9cf5c059a4f4f0cd2b974861043381664e7102e3783f48842891d43R388

It won't be usable until we develop the egress feature for user defined primary networks.

**Comment by tssurya** [comment]
This is being brought in via https://github.com/ovn-org/ovn-kubernetes/pull/4557

**Pull request closed without merging**